src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
```

Comment:
Do we want to make `origin` optional? e.g. default to `Vector(0, 0, 0, 'cm')`?
Can we ever have a case here one group has positions in `cm` and another group has positions in `pc`? The minus operation would fail, but maybe I am being too careful?

Reply:
> Do we want to make `origin` optional? e.g. default to `Vector(0, 0, 0, 'cm')`?

I'd rather not have it be optional.

> Can we ever have a case here one group has positions in `cm` and another group has positions in `pc`? The minus operation would fail, but maybe I am being too careful?

I'm not sure I follow, why would the operation fail? It's applied on `Vector`, which handles the unit conversions for us.